sdk/kubeflow/trainer/api/trainer_client.py

```diff
@@ -153,20 +153,21 @@ def train(
         self,
         runtime: types.Runtime = types.DEFAULT_RUNTIME,
         initializer: Optional[types.Initializer] = None,
-        trainer: Optional[types.CustomTrainer] = None,
+        trainer: Union[types.CustomTrainer, types.BuiltinTrainer, None] = None,
```

Comment:
```suggestion
        trainer: Optional[Union[types.CustomTrainer, types.BuiltinTrainer]] = None,
```

Could you use the `Optional` types?

Reply:
Done in [92e9555](https://github.com/kubeflow/trainer/pull/2646/commits/92e9555660d3b7bfd829594181024803613fd72f)